core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -831,9 +876,9 @@ object UnitActions {
 
     fun canPillage(unit: MapUnit, tile: TileInfo): Boolean {
         if (unit.isTransported) return false
-        val tileImprovement = tile.getTileImprovement()
+        val tileImprovement = tile.getUnpillagedTileImprovement()
         // City ruins, Ancient Ruins, Barbarian Camp, City Center marked in json
-        if (tileImprovement == null || tileImprovement.hasUnique(UniqueType.Unpillagable)) return false
+        if ((tileImprovement == null || tileImprovement.hasUnique(UniqueType.Unpillagable)) && tile.getUnpillagedRoad() == RoadStatus.None) return false
```

Comment:
Doesn't this still have the same problem as last PR? I.e., if you try to pillage a tile with both a road and an unpillagable improvement, you still pillage the improvement?

Reply:
> > NOT REVERSE COMPATIBLE
> 
> If this is the case, please update the `CURRENT_COMPATABILITY_NUMBER` variable in the companion object of `GameInfo` (`GameInfo.kt:71`) to enforce this :)

This is for marking that saves will break if you try to open one with the features in an older version?

Reply:
tileInfo.setPillaged() handles Unpillagable now

Reply:
Ah my bad, missed that.